src/translators/sql_translator.py

```diff
@@ -305,9 +316,10 @@ def _insert_entities_of_type(self,
 
         # Create/Update metadata table for this type
         table_name = self._et2tn(entity_type, fiware_service)
-        self._update_metadata_table(table_name, original_attrs)
+        modified = self._update_metadata_table(table_name, original_attrs)
         # Sort out data table.
-        self._prepare_data_table(table_name, table, fiware_service)
+        if modified:
+            self._prepare_data_table(table_name, table, fiware_service)
```

Comment:
now `_update_metadata_table` return 0 if no difference between existing metadata and new ones, if difference it return 1, this allows to avoid useless calls to `_prepare_data_table`